crates/bevy_ecs/src/world/world_cell.rs

```diff
@@ -183,8 +183,11 @@ impl<'w> WorldCell<'w> {
     /// Gets a reference to the resource of the given type
     pub fn get_resource<T: Resource>(&self) -> Option<WorldBorrow<'_, T>> {
         let component_id = self.world.components.get_resource_id(TypeId::of::<T>())?;
-        let resource_archetype = self.world.archetypes.resource();
-        let archetype_component_id = resource_archetype.get_archetype_component_id(component_id)?;
+        let archetype_component_id = self
```

Comment:
This bit of logic is duplicated repeatedly. Should we make an internal function for it?

Reply:
Done.